extensions/Worker.Extensions.Storage.Blobs/src/BlobStorageConverter.cs

```diff
@@ -160,47 +118,120 @@ private bool IsBlobExtension(ModelBindingData bindingData)
                 throw new ArgumentNullException(nameof(containerName));
             }
 
-            return await ToTargetTypeAsync(targetType, connectionName, containerName, blobName);
+            BlobContainerClient container = CreateBlobContainerClient(connectionName, containerName);
+
+            if (IsCollectionBinding(blobName, targetType))
+            {
+                Type elementType = targetType.IsArray ? targetType.GetElementType() : targetType.GenericTypeArguments[0];
+
+                if (elementType == typeof(BlobContainerClient))
+                {
+                    throw new InvalidOperationException("Collections of BlobContainerClient are not supported.");
+                }
+
+                var resultList = new List<object>();
```

Comment:
@fabiocav so this used to be `IList result = Array.CreateInstance(elementType, collectionModelBindingData.ModelBindingDataArray.Length);`

But we don't know the length here and I would like to avoid iterating over `resultSegment` just to get a count. Do you have any suggestions for a clean way to do this?

Everything is working at the moment up till `result = (IList)Activator.CreateInstance(resultType, resultList);`